### ISSUE TITLE
Controlbox disappears on logout for an untrusted device

### ISSUE THREAD

**Issue by chrzaszcz** [conversejs/converse.js#3268]
**Describe the bug**
When the use unchecks "This is a trusted device", the storage is cleaned up on logout. However, the whole controlbox is lost in the process, making the whole Web UI empty, i.e. the whole login screen disappears.

**To Reproduce**
1. Uncheck "This is a trusted device"
2. Logout
3. UI is empty

**Expected behavior**
Login screen should appear again.

**Screenshots**
Completely blank UI, so no screenshot needed.

**Environment (please complete the following information):**
 - Desktop
 - Latest Firefox.
 - Converse.js 10.1.6

**Additional context**
I spent some time debugging this, and I found that commenting out the following lines resolves the issue:

```javascript
export async function onClearSession () {
    if (shouldClearCache()) {
        await Promise.all(
            _converse.chatboxes.map(c => c.messages && c.messages.clearStore({ 'silent': true }))
        );
        //const filter = o => o.get('type') !== _converse.CONTROLBOX_TYPE;
        //_converse.chatboxes.clearStore({ 'silent': true }, filter);
    }
}
```

It looks like the `clearStore` function in `skeletor` is supposed to take the filter into consideration, and only clean up the boxes that are not filtered out. However, it calls `this._reset()` which deletes all models, making the controlbox disappear when the UI is later updated.

Is my reasoning correct?

**Comment by jcbrand** [comment]
> Is my reasoning correct?

Yes I think it's correct. I think the fix is to update `clearStore` in Skeletor to pass the filtered out model(s) to `this.reset()` so that those models are kept.

See here: https://github.com/conversejs/skeletor/blob/69b49b4cfcb2db323ac0bed5472ebb8cd714b6c5/src/collection.js#L277
